test/rules/implicit_reopen_test.dart

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../rule_test_support.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(ImplicitReopenTest);
+    defineReflectiveTests(ImplicitReopenInducedModifierTest);
+  });
+}
+
+@reflectiveTest
+class ImplicitReopenInducedModifierTest extends LintRuleTest
+    with LanguageVersion300Mixin {
+  @override
+  bool get addMetaPackageDep => true;
+
+  @override
+  String get lintRule => 'implicit_reopen';
+
+  test_class_classSealed_classFinal_ok() async {
```

Comment:
Nit: The naming of this test is very unlike the others heh

Reply:
Ha. Yeah. And all awkward!  (Made it a bit better.) Thanks!